packages/cli/src/util/paths.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  serializeWithPipeDelimitedStyle,
+  serializeWithSpaceDelimitedStyle,
+} from '@stoplight/prism-http/src/mocker/serializer/style/delimited';
+import {
+  Dictionary,
+  HttpParamStyles,
+  IHttpOperation,
+  IHttpParam,
+  IHttpPathParam,
+  IHttpQueryParam,
+  INodeExample,
+  INodeExternalExample,
+} from '@stoplight/types';
+import * as Array from 'fp-ts/lib/Array';
+import * as Either from 'fp-ts/lib/Either';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { get } from 'lodash';
+// @ts-ignore
+import { parse } from 'uri-template';
+
+export function createExamplePath(operation: IHttpOperation): Either.Either<Error, string> {
+  return pipe(
+    generateTemplateAndValuesForPathParams(operation),
+    Either.chain(({ template: pTemplate, values: pValues }) => {
+      return pipe(
+        generateTemplateAndValuesForQueryParams(pTemplate, operation),
+        Either.map(({ template: qTemplate, values: qValues }) => {
+          return { template: qTemplate, values: { ...pValues, ...qValues } };
+        }),
+      );
+    }),
+    Either.map(({ template, values }) => parse(template).expand(values)),
+  );
+}
+
+function generateParamValue(spec: IHttpParam): Either.Either<Error, unknown> {
+  return pipe(
+    generate(spec),
+    Either.fromOption(() => new Error(`Cannot generate value for: ${spec.name}`)),
+    Either.chain(value => {
+      switch (spec.style) {
+        case HttpParamStyles.DeepObject:
+          return serializeWithDeepObjectStyle(spec.name, value);
+
+        case HttpParamStyles.PipeDelimited:
+          return serializeWithPipeDelimitedStyle(spec.name, value as Array<string | number | boolean>, spec.explode);
+
+        case HttpParamStyles.SpaceDelimited:
+          return serializeWithSpaceDelimitedStyle(spec.name, value as Array<string | number | boolean>, spec.explode);
+
+        default:
+          return Either.right(value);
+      }
+    }),
+  );
+}
+
+function generateParamValues(specs: IHttpParam[]) {
+  return specs.reduce((valuesOrError: Either.Either<Error, Dictionary<unknown, string>>, spec) => {
+    return pipe(
+      valuesOrError,
+      Either.chain(values =>
+        pipe(
+          generateParamValue(spec),
+          Either.map(value => ({
+            ...values,
+            [spec.name]: value,
+          })),
+        ),
+      ),
+    );
+  }, Either.right({}));
+}
+
+function generateTemplateAndValuesForPathParams(operation: IHttpOperation) {
+  const specs = get(operation, 'request.path', []);
+
+  return pipe(
+    generateParamValues(specs),
+    Either.chain(values => {
+      return pipe(
+        createPathUriTemplate(operation.path, specs),
+        Either.map(template => ({ template, values })),
+      );
+    }),
+  );
+}
+
+function generateTemplateAndValuesForQueryParams(template: string, operation: IHttpOperation) {
+  const specs = get(operation, 'request.query', []);
```

Comment:
This is up to you, but I usually prefer make this function accept a `QueryParameter[]` directly instead of getting the whole `IHttpOperation`, get the value from the property and then use it.

Up to you though, it's not blocking anything.

Reply:
It is like that just for `createExamplePath` (entry point) to look more balanced. I'd vote on leaving it like that instead of moving that dirty lodash's get to our entry point.

Reply:
Forget ^. I exported it from http package.